targets/TARGET_STM/stm_spi_api.c

```diff
@@ -349,22 +350,54 @@ static inline int ssp_busy(spi_t *obj)
 
 int spi_master_write(spi_t *obj, int value)
 {
-    uint16_t size, ret;
-    int Rx = 0;
     struct spi_s *spiobj = SPI_S(obj);
     SPI_HandleTypeDef *handle = &(spiobj->handle);
 
-    size = (handle->Init.DataSize == SPI_DATASIZE_16BIT) ? 2 : 1;
+#ifdef STM32_SPI_KEEP_HAL_USAGE
```

Comment:
The commit message describes it: 

This commit implements a SPI mode which will offer better performance
thanks to usage of Lower Layer API which use fewer registers access,
at the cost of lower robustness (no error management at all).

The SPI driver can be configured to re-use the HAL and error management by
defining the STM32_SPI_KEEP_HAL_USAGE compilation switch